server/game/cards/02_SHD/units/ValiantAssaultShip.ts

```diff
@@ -0,0 +1,35 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { RelativePlayer, WildcardCardType, ZoneName } from '../../../core/Constants';
+
+export default class ValiantAssaultShip extends NonLeaderUnitCard {
+    protected override getImplementationId () {
+        return {
+            id: '7922308768',
+            internalName: 'valiant-assault-ship'
+        };
+    }
+
+    public override setupCardAbilities () {
+        this.addOnAttackAbility({
+            title: 'If the defending player controls more resources than you, this unit gets +2/+0 for this attack',
+            targetResolver: {
+                controller: RelativePlayer.Self,
+                cardTypeFilter: WildcardCardType.Unit,
+                zoneFilter: ZoneName.SpaceArena,
+                immediateEffect: AbilityHelper.immediateEffects.conditional({
+                    condition: (context) => {
+                        return context.source.controller.resources.length < context.source.controller.opponent.resources.length;
```

Comment:
Can do this all in one line as:

```typescript:
condition: (context) => context.source.controller.resources.length < context.source.controller.opponent.resources.length
```

Reply:
yep, good catch, that was left over from debugging